src/main/java/org/openrewrite/kotlin/internal/KotlinParserVisitor.java

```diff
@@ -1815,64 +1820,58 @@ public J visitProperty(FirProperty property, ExecutionContext ctx) {
-                }
-                saveCursor = cursor;
-                whitespace();
+                    LeafPsiElement accessorNode = getGetterOrSetterNode(ktPropertyAccessor);
+                    if (accessorNode == null) {
+                        throw new RuntimeException("a ktPropertyAccessor node should have get or set");
+                    }
 
-                methodName = source.substring(cursor, cursor + 3);
-                switch (methodName) {
-                    case "get":
-                        cursor(saveCursor);
-                        if (isValidGetter(property.getGetter())) {
-                            getter = (J.MethodDeclaration) visitElement(property.getGetter(), ctx);
-                            if (receiver != null) {
-                                getter = getter.withParameters(ListUtils.concat(receiver, getter.getParameters()));
-                            }
-                            expressions.add(getter);
-                        }
-                        break;
-                    case "set":
-                        cursor(saveCursor);
-                        if (isValidSetter(property.getSetter())) {
-                            setter = (J.MethodDeclaration) visitElement(property.getSetter(), ctx);
-                            if (receiver != null) {
-                                setter = setter.withParameters(ListUtils.concat(receiver, setter.getParameters()));
-                            }
-                            expressions.add(setter);
-                        }
-                        break;
-                    default:
-                        cursor(saveCursor);
-                        break;
+                    Space accessorPrefix = EMPTY;
+                    if (accessorNode.getPrevSibling() instanceof PsiWhiteSpace) {
+                        accessorPrefix = Space.format(accessorNode.getPrevSibling().getText());
+                    }
+
+                    FirPropertyAccessor firPropertyAccessor = null;
+                    if (accessorNode.getText().equals("get")) {
+                        firPropertyAccessor = property.getGetter();
+                    } else if (accessorNode.getText().equals("set")) {
+                        firPropertyAccessor = property.getSetter();
+                    }
+
+                    J.MethodDeclaration m = (J.MethodDeclaration) visitElement(firPropertyAccessor, ctx);
+                    if (receiver != null) {
+                        m = m.withParameters(ListUtils.concat(receiver, m.getParameters()));
+                    }
+                    if (hasAnnotationBeforeAccessors) {
+                        m = m.withLeadingAnnotations(annos)
+                            .withName(m.getName().withPrefix(accessorPrefix))
+                            .withPrefix(EMPTY);
+                    }
+                    expressions.add(m);
+
+                    cursor(ktPropertyAccessor.getTextRange().getEndOffset());
```

Comment:
I think setting the cursor based on the PSI may cause code to be skipped if we don't parse the elements perfectly.
Are you sure it's safe to set the cursor based on the text range?
Is there a more reliable way to increment the cursor?